StatefulSet partition value update doesn't force server redeployment so it shouldn't block acl bootstrapper job

I like the work that's been done on the bootstrap-acls job but I'm running into one quirk.
When the partition value of the statefulset is upgraded but nothing else in the pod spec, the servers do not redeploy.

This perceived difference in the partition value causes the acl-init job to get stuck indefinitely with the following repeating message:
```
2019-10-28T20:30:16.131Z [ERROR] Failure: waiting for rollout of statefulset vault-consul-consul-server: err="rollout is in progress (CurrentRevision=vault-consul-consul-server-844f64bff6 UpdateRevision=vault-consul-consul-server-86c7b4f99)"
2019-10-28T20:30:16.131Z [INFO ] Retrying in 1s
2019-10-28T20:30:17.135Z [ERROR] Failure: waiting for rollout of statefulset vault-consul-consul-server: err="rollout is in progress (CurrentRevision=vault-consul-consul-server-844f64bff6 UpdateRevision=vault-consul-consul-server-86c7b4f99)"
2019-10-28T20:30:17.135Z [INFO ] Retrying in 1s
...
``` 
In my opinion, the partition value shouldn't block the acl-init job and instead a new condition should be looked for.
Alternative 1: add logic to the job to detect that a rollout isn't actually occurring and that the job should continue (eg: no server redeployments in at least a minute).
Alternative 2: add something to the Consul server (eg: an env) to force redeployment of the consul servers when the partition value is updated.

--- a/templates/server-acl-init-cleanup-job.yaml
+++ b/templates/server-acl-init-cleanup-job.yaml
@@ -1,5 +1,7 @@
 {{- if (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) }}
 {{- if .Values.global.bootstrapACLs }}
+{{- /* See reason for this in server-acl-init-job.yaml */ -}}
+{{- if eq (int .Values.server.updatePartition) 0 }}
 # This job deletes the server-acl-init job once it completes successfully.
 # It runs as a helm hook because it only needs to run when the server-acl-init
 # Job gets recreated which only happens during an install or upgrade.
@@ -49,5 +51,6 @@ spec:
             - delete-completed-job
             - -k8s-namespace={{ .Release.Namespace }}
             - {{ template "consul.fullname" . }}-server-acl-init
+  {{- end }}
   {{- end }}
   {{- end }}

--- a/templates/server-acl-init-job.yaml
+++ b/templates/server-acl-init-job.yaml
@@ -1,5 +1,13 @@
 {{- if (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) }}
 {{- if .Values.global.bootstrapACLs }}
+{{- /* We don't render this job when server.updatePartition > 0 because that
+    means a server rollout is in progress and this job won't complete unless
+    the rollout is finished (which won't happen until the partition is 0).
+    If we ran it in this case, then the job would not complete which would cause
+    the server-acl-init-cleanup hook to run indefinitely which would cause the
+    helm upgrade command to hang.
+*/ -}}
+{{- if eq (int .Values.server.updatePartition) 0 }}
 apiVersion: batch/v1
 kind: Job
 metadata:
@@ -64,5 +72,6 @@ spec:
                 -create-client-token=false \
                 {{- end }}
                 -expected-replicas={{ .Values.server.replicas }}
+{{- end }}
 {{- end }}
 {{- end }}

--- a/test/unit/server-acl-init-cleanup-job.bats
+++ b/test/unit/server-acl-init-cleanup-job.bats
@@ -43,6 +43,17 @@ load _helpers
   [ "${actual}" = "true" ]
 }
 
+@test "serverACLInitCleanup/Job: disabled when server.updatePartition > 0" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/server-acl-init-cleanup-job.yaml  \
+      --set 'global.bootstrapACLs=true' \
+      --set 'server.updatePartition=1' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+
 @test "serverACLInitCleanup/Job: consul-k8s delete-completed-job is called with correct arguments" {
   cd `chart_dir`
   local actual=$(helm template \

--- a/test/unit/server-acl-init-job.bats
+++ b/test/unit/server-acl-init-job.bats
@@ -43,6 +43,17 @@ load _helpers
   [ "${actual}" = "true" ]
 }
 
+@test "serverACLInit/Job: disabled when server.updatePartition > 0" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/server-acl-init-job.yaml  \
+      --set 'global.bootstrapACLs=true' \
+      --set 'server.updatePartition=1' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+
 @test "serverACLInit/Job: does not set -create-client-token=false when client is enabled (the default)" {
   cd `chart_dir`
   local actual=$(helm template \

